### PR TITLE
Cafeteria Scheduler Front End

### PR DESCRIPTION
**Pull request by Tcintra**
Implement UI to communicate with the CafeteriaScheduler class using a CafeteriaSchedulerServlet. Additional servlets handle preferences and school inputs and store them in the datastore. The cafeteria schedule is stored as an embedded entity of the Cafeteria Entity

### CODE REVIEW

**Review by rivera-fernando — APPROVED** [review]

**Review by christinachance — APPROVED** [review]

**Review by christinachance — APPROVED** [review]
Add empty line at end of files, not sure which one but I think there is a file.

**Review by christinachance — APPROVED** [review]

File: project/src/main/java/com/google/sps/servlets/LogInServlet.java [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/servlets/LogInServlet.java]: Unused imports?

File: project/src/main/java/com/google/sps/servlets/LogInServlet.java [hunk 2]
```diff
@@ -63,6 +65,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             ArrayList<String> users = new ArrayList<String>();
             for (Entity entity:results.asIterable()){
                 if (entity.getProperty("password").equals(password) && entity.getProperty("email").equals(email)){
+
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/servlets/LogInServlet.java]: remove empty line

File: project/src/main/java/com/google/sps/servlets/SignUpServlet.java [hunk 3]
```diff
@@ -3,6 +3,8 @@
 import com.google.appengine.api.images.ServingUrlOptions;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
```
Comment (serzhk) [inline project/src/main/java/com/google/sps/servlets/SignUpServlet.java]: Unused imports?

File: project/src/main/webapp/event_script.js [hunk 4]
```diff
@@ -640,4 +640,3 @@ function validateEvent() {
   }
   return true;
 }
-
```
Comment (serzhk) [inline project/src/main/webapp/event_script.js:643]: Revert back changes in this file

File: project/src/main/java/com/google/sps/classes/Cafeteria.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * Class representing an individual event.
+ *
+ */
+public final class Cafeteria {
+ 
+  private final String school;
+  private final String name;
+  private final int maxCapacity;
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/classes/Cafeteria.java]: the var should be `max_capacity` to be consistent with other variables

File: project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+        long dinner_start = (long) entity.getProperty("dinner_start");
+        long dinner_end = (long) entity.getProperty("dinner_end");
+
+        TimeRange lunch_time = TimeRange.fromStartEnd(Math.toIntExact(lunch_start), Math.toIntExact(lunch_end), false);
+        TimeRange dinner_time = TimeRange.fromStartEnd(Math.toIntExact(dinner_start), Math.toIntExact(dinner_end), false);
+
+        for (int index = 0; index < studentResultsList.size(); index++) {
+          Entity studentEntity = studentResultsList.get(index);
+          if (studentEntity.getProperty("cafeteria").equals(cafeteriaName)) {
+          long lunch_start_pref = (long) studentEntity.getProperty("lunch_start");
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java]: indentation

Comment (Tcintra) [reply project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java]: Indented!

File: project/src/main/java/com/google/sps/classes/Cafeteria.java (hunk 5 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/classes/Cafeteria.java]: Changed it to max_capacity and meal_time, not sure why I used different naming patterns :(

File: project/src/main/java/com/google/sps/servlets/SignUpServlet.java (hunk 3 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/servlets/SignUpServlet.java]: Deleted them! I was going to create a new property and then changed my mind but forgot to remove the imports..

File: project/src/main/java/com/google/sps/servlets/LogInServlet.java (hunk 1 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/servlets/LogInServlet.java]: Removed!

File: project/src/main/java/com/google/sps/servlets/LogInServlet.java (hunk 2 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/servlets/LogInServlet.java]: Removed!

File: project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+        }
+        
+        scheduleEntity.setProperty("lunch_blocks", lunchBlocksEntity);
+        scheduleEntity.setProperty("dinner_blocks", dinnerBlocksEntity);
+        entity.setProperty("schedule", scheduleEntity);
+        entity.setProperty("is_scheduled", true);
+        datastore.put(entity);
+      } catch(EntityNotFoundException e) {
+        e.printStackTrace();
+      }
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java:143]: line 120-152 could be a function like `setMealEntity`
```
public EmbeddedEntity setMealEntity(mealEntity, mealBlock ){
for (Block block : schedule.getDinnerBlocks()) {
          mealBlock.setProperty("start", block.getTime().start());
          mealBlock.setProperty("end", block.getTime().end());
          List<String> blockStudents = new ArrayList<>();
          for (Student blockStudent : block.getStudents()) {
            blockStudents.add(blockStudent.getName());
          }
          mealBlock.setProperty("students", blockStudents);
          mealEntity.setProperty(blockName.concat(Integer.toString(blockIndex)), mealBlock);
          blockIndex++;
        }
}
return mealEntity;
```

File: project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+
+      int lunch_start = Math.toIntExact((long) cafeteria.getProperty("lunch_start"));
+      int lunch_end = Math.toIntExact((long) cafeteria.getProperty("lunch_end"));
+      int dinner_start = Math.toIntExact((long) cafeteria.getProperty("dinner_start"));
+      int dinner_end = Math.toIntExact((long) cafeteria.getProperty("dinner_end"));
+
+      int numLunchBlocks = (lunch_end-lunch_start)/mealTime;
+      int numDinnerBlocks = (dinner_end-dinner_start)/mealTime;
+
+      for (int lunchIndex = 0; lunchIndex < numLunchBlocks; lunchIndex++) {
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java]: same above, should have a function to make line 190-210 more clean.

**Review by wulian1223 — APPROVED** [review]
LGTM after fix comments

File: project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java (hunk 7 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java:143]: done!

File: project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java (hunk 8 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/servlets/CafeteriaSchedulerServlet.java]: done!

**Pull request merged**